core/src/test/java/org/apache/calcite/test/SqlValidatorBestEffortTest.java

```diff
@@ -91,20 +91,20 @@
   @Test public void testUnknownTableMerge() {
     String sql = "MERGE INTO foo AS f USING dept AS b ON f.bar = b.name"
```

Comment:
Since UnknownRecordType objects are now connected to an underlying schema, each time a field is queried within them, that field gets stored within the record (in other words, if you have an unknown table and you query whether it contains column abc, the table will return yes, and will internally record that one of the fields it contains is column abc). This meant that I had to tweak existing tests to account for this behavior.